src/lib/videotagging_extensions/detection.js

```diff
@@ -0,0 +1,193 @@
+
+        self.detectionAlgorithmManager.setExporter(method, exportPath, self.videotagging.inputtagsarray, 
+                                            Object.keys(self.videotagging.frames).length,
+                                            self.videotagging.video.videoWidth,
+                                            self.videotagging.video.videoHeight,
+                                            testSplit);
+
+        self.detectionAlgorithmManager.exporter.init()
+            .then(() => {
+                this.mapVideo(exportFrame,exportUntil).then(() => {
+                let notification = new Notification('Offline Video Tagger', {
+                    body: 'Successfully exported CNTK files.'
+                });
+                cb();
+            })  
+            }, (err) => {
+                console.info(`Error on ${self.detectionAlgorithmManager.getCurrentExportAlgorithm()} init:`, err);
+                cb(err)
+            });
+
+        function exportFrame(frameId, frameCanvas, canvasContext, frameExportCb) {
+            if (!self.visitedFrames.has(frameId)) {
+                return frameExportCb();
+            }
+            var frameTags = [];
+            //confirm that frame is tagged and that no tags are unlabeled 
+            var frameIsTagged = self.videotagging.frames.hasOwnProperty(frameId) && (self.videotagging.frames[frameId].length);
+            if (frameIsTagged && (self.videotagging.getUnlabeledRegionTags(frameId).length != self.videotagging.frames[frameId].length)) {
+                //genereate metadata from tags
+                self.videotagging.frames[frameId].map( (tag) => {
+                    if (!tag.tags[tag.tags.length-1]) {
+                        return console.log(`frame ${frameId} region ${tag.name} has no label`);
+                    }
+                    var stanW = self.videotagging.video.videoWidth/tag.width;
+                    var stanH = self.videotagging.video.videoHeight/tag.height;
+                    frameTags.push({
+                        class : tag.tags[tag.tags.length-1],
+                        x1 : parseInt(tag.x1 * stanW),
+                        y1 : parseInt(tag.y1 * stanH),
+                        x2 : parseInt(tag.x2 * stanW),
+                        y2 : parseInt(tag.y2 * stanH)
+                    });
+
+                });
+            }
+
+            //draw the frame to the canvas
+            canvasContext.drawImage(self.videotagging.video, 0, 0);
+            var data = frameCanvas.toDataURL('image/jpeg').replace(/^data:image\/\w+;base64,/, ""); // strip off the data: url prefix to get just the base64-encoded bytes http://stackoverflow.com/questions/5867534/how-to-save-canvas-data-to-file
+            var buf = new Buffer(data, 'base64');
+            var frameFileName = `${pathJS.basename(self.videotagging.src, pathJS.extname(self.videotagging.src))}_frame_${frameId}.jpg`
```

Comment:
Please use node's path module instead of PathJS (I assume this is a leftover from a UI code)